app/models/purchase.rb

```diff
@@ -3183,14 +3201,22 @@ def calculate_fees
       self.affiliate_credit_cents = determine_affiliate_balance_cents
     end
 
+    def seller_based_discover_fee_per_thousand
+      if seller&.custom_discover_fee_percentage.present?
+        seller.custom_discover_fee_percentage / 100.0 * 1000
+      else
+        GUMROAD_DISCOVER_FEE_PER_THOUSAND
+      end
+    end
+
     def calculate_additional_discover_fee_per_thousand
       if is_recurring_subscription_charge || is_updated_original_subscription_purchase
         subscription.original_purchase.discover_fee_per_thousand - (flat_fee_applicable? ? GUMROAD_DISCOVER_EXTRA_FEE_PER_THOUSAND : 0) - (subscription.mor_fee_applicable? && charged_using_gumroad_merchant_account? ? PROCESSOR_FEE_PER_THOUSAND : 0)
       elsif is_preorder_charge?
         preorder.authorization_purchase.discover_fee_per_thousand - (flat_fee_applicable? ? GUMROAD_DISCOVER_EXTRA_FEE_PER_THOUSAND + PROCESSOR_FEE_PER_THOUSAND : 0)
       else
         if Feature.active?(:merchant_of_record_fee, seller)
-          GUMROAD_DISCOVER_FEE_PER_THOUSAND - GUMROAD_DISCOVER_EXTRA_FEE_PER_THOUSAND - (charged_using_gumroad_merchant_account? ? PROCESSOR_FEE_PER_THOUSAND : 0)
+          [seller_based_discover_fee_per_thousand - GUMROAD_DISCOVER_EXTRA_FEE_PER_THOUSAND - (charged_using_gumroad_merchant_account? ? PROCESSOR_FEE_PER_THOUSAND : 0), 0].max
```

Comment:
Here, I am unsure of the logic as in we are subtracting Extra fee and processor fee from discover fee. To make sure it does not go to negative, I am using max(,0)